### PR TITLE
[2.11.x] DDF-3369 Fixed ability to delete DeletedMetacards

### PR DESCRIPTION
**Pull request by rzwiefel**
#### What does this PR do?
Fixed ability to delete DeletedMetacards and also fixes SecurityLogger in the standardframework bundle

#### Who is reviewing it? 
anyone
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
anyone
#### How should this be tested? (List steps with links to updated documentation)
ensure you can restore metacards and the DeletedMetacard marker gets deleted on restore
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-3369
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java [hunk 1]
```diff
@@ -108,6 +94,14 @@
   private static final Integer DEFAULT_MAX_PAGE_SIZE = 1000;
 
   public static final Integer MAX_PAGE_SIZE = determineAndRetrieveMaxPageSize();
+  // Inject properties
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java]: ❓ Did you make all these formatting changes by hand, or through the formatter plugin?

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java [hunk 2]
```diff
@@ -765,12 +765,15 @@ private void attemptDeleteDeletedMetacard(String id)
       LOGGER.debug("There should have been one deleted metacard marker");
       return;
     }
-    final QueryResponse _response = response;
+
+    final DeleteRequestImpl deleteRequest =
+        new DeleteRequestImpl(response.getResults().get(0).getMetacard().getId());
+    deleteRequest
+        .getProperties()
+        //        .put("operation.query-tags", ImmutableSet.of(DeletedMetacard.DELETED_TAG));
```
Comment (coyotesqrl) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java]: ✏️ Comment left in why?

Comment (clockard) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java]: Remove commented out line

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java (hunk 1 above)
Comment (rzwiefel) [reply catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java]: By accident I believe. I think i accidentally checked rearrange code

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java (hunk 2 above)
Comment (rzwiefel) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java]: done

**Comment by rzwiefel** [comment]
@coyotesqrl  Unless you want a unit test that walks the Filter tree, no unit tests to add.

**Comment by clockard** [comment]
build now

**Comment by rzwiefel** [comment]
build now

**Comment by rzwiefel** [comment]
build now

**Comment by rzwiefel** [comment]
build now

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java [hunk 3]
```diff
@@ -659,25 +671,29 @@ private QueryResponse addProcessingDetails(
...
-                        .is()
-                        .like()
-                        .text(DeletedMetacard.DELETED_TAG)));
+  @Nullable
+  protected Filter getNonVersionTagsFilter(Operation requestOperation) {
+    FilterBuilder filterBuilder = frameworkProperties.getFilterBuilder();
+    if (requestOperation.containsPropertyName("operation.query-tags")) {
+      Set<String> queryTags =
+          Optional.of(requestOperation)
+              .map((ro) -> ro.getPropertyValue("operation.query-tags"))
```
[the remark this answers is not in the document]
Comment (rzwiefel) [reply catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java]: Thanks, but no thanks.

**Comment by rzwiefel** [comment]
build now

**Comment by rzwiefel** [comment]
build now

**Comment by rzwiefel** [comment]
build now

**Comment by clockard** [comment]
build now

**Comment by rzwiefel** [comment]
build now

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
Manually tested and verified. One note is that archived entries can be cached so after a result is restored it might still show up in an archive search.

**Comment by clockard** [comment]
build now

**Pull request merged**